### ISSUE TITLE
JSON accepted with single quote

### ISSUE THREAD

**Issue by smhnaji** [FreeCodeCamp/FreeCodeCamp#9052]
Challenge [Introducing JavaScript Object Notation JSON](https://www.freecodecamp.com/challenges/introducing-javascript-object-notation-json#?solution=%0Avar%20myMusic%20%3D%20%5B%0A%20%20%7B%0A%20%20%20%20%22artist%22%3A%20%22Billy%20Joel%22%2C%0A%20%20%20%20%22title%22%3A%20%22Piano%20Man%22%2C%0A%20%20%20%20%22release_year%22%3A%201973%2C%0A%20%20%20%20%22formats%22%3A%20%5B%20%0A%20%20%20%20%20%20%22CS%22%2C%20%0A%20%20%20%20%20%20%228T%22%2C%20%0A%20%20%20%20%20%20%22LP%22%20%5D%2C%0A%20%20%20%20%22gold%22%3A%20true%0A%20%20%7D%2C%0A%20%20%7B%0A%20%20%20%20%22artist%22%20%3A%20%22asdas%22%2C%0A%20%20%20%20%22title%22%20%3A%20%22ti4l%22%2C%0A%20%20%20%20%22release_year%22%3A%201998%2C%0A%20%20%20%20%22formats%22%3A%20%5B%27a%27%2C%20%22b%22%2C%20%22c%22%5D%0A%20%20%7D%0A%20%20%2F%2F%20Add%20record%20here%0A%5D%3B%0A%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:46.0) Gecko/20100101 Firefox/46.0</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:

``` javascript

var myMusic = [
  {
    "artist": "Billy Joel",
    "title": "Piano Man",
    "release_year": 1973,
    "formats": [ 
      "CS", 
      "8T", 
      "LP" ],
    "gold": true
  },
  {
    "artist" : "asdas",
    "title" : "ti4l",
    "release_year": 1998,
    "formats": ['a', "b", "c"] ////////////// Consider single-quotes around 'a'
  }
  // Add record here
];


```

As far as I know (and have researched), single quotes are not permitted when it comes to JSON. (I had also had serious problems with it before!).

So I think...
1. It's good to mention that only double quotes are permitted.
2. Please raise an error if the input was given by the trainer like what I tested above.

Thank you

**Comment by zackharley** [comment]
So the issue here is that the objects in the array are [JavaScript Object Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer), but the student is being told they are JSON. Since these are Object Literals, that means that your `myMusic` object is valid and should not throw an error when created. 

As long as the Object Literal that is created is a valid Object, then it can be converted into a JSON string:
```
JSON.stringify(myMusic); // The output would be of type String with 'a' changed to "a"
```

If you want to know more about the difference between JSON and Object Literals, [here is a link to a Stack Overflow question](http://stackoverflow.com/questions/2904131/what-is-the-difference-between-json-and-object-literal-notation).

**Comment by erictleung** [comment]
@smhnaji agreed. The [JSON official website](http://json.org/) only uses double quotes:

> A value can be a string in double quotes, or a number, or true or false or null, or an object or an array. 

For more evidence, the [jQuery API](http://api.jquery.com/jQuery.parseJSON/) only takes double quotes:

> `"{'test': 1}"` ('test' is using single quotes instead of double quotes).

And I agree with your suggestions:

> 1. It's good to mention that only double quotes are permitted.
> 2. Please raise an error if the input was given by the trainer like what I tested above.

Related to #7767 and #7767.

**Comment by josectello** [comment]
Looks like we would need to add in another test case for enforcing double quotes and perhaps update the challenge description to mention JSON formatting?

**Comment by crashuniverse** [comment]
I think reference to #7767 is really good and the discussion should move there. #7767 captures the essence of underlying problem, using the term JSON incorrectly. If we fix doublequotes, we'd just create more confusion because these are JavaScript collections.

**Comment by systimotic** [comment]
As #7767 is resolved and this challenge no longer references JSON, can this issue be closed?

**Comment by smhnaji** [comment]
Yes, please.

Thank you for the follow up :)

**Comment by erictleung** [comment]
@systimotic @smhnaji thanks for looking into the issue. Yeah, looks like this issue is fixed with #9144. Closing as resolved. Happy coding, all!
